### DISCUSSION TITLE
[Question] Docker-gen stopping randomly

### DISCUSSION THREAD

**Discussion by NanoCode012** [evertramos/nginx-proxy-automation#338] [Q&A] [answered]
Hello, I came across a situation where `docker-gen` would abruptly stop with Exit code 2.

This has been the second time it happened in the past few months. I tried to check `docker-gen`'s logs and it does not show anything at the end, so I could not provide much details or recreate it.

I was wondering whether others encountered this issue as well? 
Version: b478147bef4a34bdfafb8a2c63e2a8a6a9eeda2e

I'm on an old commit, but I don't see any Issue about this.

**Comment by evertramos** [comment, marked as answer]
Hello @NanoCode012, thanks for posting.

I myself had this issue a long ago with previous version and I am did not dig into to find out the real reason, but I updated to the latest version of docker-gen and nginx-proxy which seems to fix this issue.

I suggest you do the same for a couple more reasons which include the new certificate issuance being updated in the latest version as well. 

If you intend to do so, please follow this guide:

https://github.com/evertramos/nginx-proxy-automation/blob/master/docs/upgrade-guide.md

**Reply by NanoCode012** [reply]
Thanks, @evertramos . I will do so now. If the issue crops back up, I'll come back here and hopefully have some logs.

**Reply by NanoCode012** [reply]
@evertramos , I just rechecked `docker-gen` repo and found that `0.8.0` fixes https://github.com/nginx-proxy/docker-gen/issues/244 which seem similar to this issue with this PR https://github.com/nginx-proxy/docker-gen/pull/378
